common/txmgr/types/fee_estimator.go

```diff
@@ -18,21 +19,23 @@ const (
 type Fee fmt.Stringer
 
 // PriorAttempt provides a generic interface for reading tx data to be used in the fee esimators
-type PriorAttempt[F Fee, HASH any] interface {
+//
+//go:generate mockery --quiet --name PriorAttempt --output ./mocks/ --case=underscore
+type PriorAttempt[F Fee, TX_HASH types.Hashable] interface {
 	Fee() F
 	GetChainSpecificGasLimit() uint32
 	GetBroadcastBeforeBlockNum() *int64
-	GetHash() HASH
+	GetHash() TX_HASH
```

Comment:
Probably out of scope for this PR, but there are lot of `Get`er methods here.

Reply:
Yes, can do it in separate PR. We do plan to do many rounds of cleanup later, which will include many such things.